pennylane/ops/op_math/controlled_ops.py

```diff
@@ -243,6 +148,105 @@ def _controlled(self, wire):
         return new_op
 
 
+class CY(ControlledOp):
+    r"""CY(wires)
+    The controlled-Y operator
+
+    .. math:: CY = \begin{bmatrix}
+            1 & 0 & 0 & 0 \\
+            0 & 1 & 0 & 0\\
+            0 & 0 & 0 & -i\\
+            0 & 0 & i & 0
+        \end{bmatrix}.
+
+    .. note:: The first wire provided corresponds to the **control qubit**.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int]): the wires the operation acts on
+    """
+    num_wires = 2
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = ()
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
+    def __init__(self, wires, do_queue=True, id=None):
+        control_wire, wire = wires
+        base = PauliY(wire)
+
+        super().__init__(base, control_wire, do_queue=do_queue, id=id)
+        self._name = "CY"
+
+    @staticmethod
+    @lru_cache()
+    def compute_matrix():  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.CY.matrix`
+
+
+        Returns:
+            ndarray: matrix
+
+        **Example**
+
+        >>> print(qml.CY.compute_matrix())
+        [[ 1.+0.j  0.+0.j  0.+0.j  0.+0.j]
+         [ 0.+0.j  1.+0.j  0.+0.j  0.+0.j]
+         [ 0.+0.j  0.+0.j  0.+0.j -0.-1.j]
+         [ 0.+0.j  0.+0.j  0.+1.j  0.+0.j]]
+        """
+        return np.array(
+            [
+                [1, 0, 0, 0],
+                [0, 1, 0, 0],
+                [0, 0, 0, -1j],
+                [0, 0, 1j, 0],
+            ]
+        )
+
+    @staticmethod
+    def compute_decomposition(wires):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators (static method).
+
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+
+        .. seealso:: :meth:`~.CY.decomposition`.
+
+        Args:
+            wires (Iterable, Wires): wires that the operator acts on
+
+        Returns:
+            list[Operator]: decomposition into lower level operations
+
+        **Example:**
+
+        >>> print(qml.CY.compute_decomposition([0, 1]))
+        [CRY(3.141592653589793, wires=[0, 1]), S(wires=[0])]
+
+        """
+        return [qml.CRY(np.pi, wires=wires), S(wires=wires[0])]
+
+    def decomposition(self):
```

Comment:
you probably don't need to re-define this method. was there a reason you added this?

Reply:
It didn't give me the correct decomposition. I.e., some tests failed.

As it stands I get:
```Python
In [2]: cy = qml.CY([0, 1])

In [3]: cy.decomposition()
Out[3]: [CRY(3.141592653589793, wires=[0, 1]), S(wires=[0])]
```

Without the custom implementation I get:
```Python
[Controlled(PhaseShift(1.5707963267948966, wires=[1]), control_wires=[0]),
 Controlled(RY(3.141592653589793, wires=[1]), control_wires=[0]),
 Controlled(PhaseShift(1.5707963267948966, wires=[1]), control_wires=[0])]
```

Reply:
ahhh `Controlled` overrides `decomposition`, so you're essentially undoing that. works for me!